### ISSUE TITLE
meta_data.json question

### ISSUE THREAD

**Issue by aledbf** [coreos/bugs#269]
After the update to 591.0.0 the nodes are not reading user_data properly.
Only using `coreos.autologin=ttyS0` I can access the console. Running `coreos-cloudinit` manually I see this:

```
core@nodo-1 ~ $ sudo coreos-cloudinit --from-configdrive=/media/configvirtfs/
Checking availability of "cloud-drive"
Fetching user-data from datasource of type "cloud-drive"
Attempting to read from "/media/configvirtfs/openstack/latest/user_data"
Fetching meta-data from datasource of type "cloud-drive"
Attempting to read from "/media/configvirtfs/openstack/latest/meta_data.json"
Failed fetching meta-data from datasource: unexpected end of JSON input
```

It's `meta_data.json` required now?
`echo "{}" > nodo1/configdrive/openstack/latest/meta_data.json`

```
core@nodo-1 ~ $ sudo coreos-cloudinit --from-configdrive=/media/configvirtfs/
Checking availability of "cloud-drive"
Fetching user-data from datasource of type "cloud-drive"
Attempting to read from "/media/configvirtfs/openstack/latest/user_data"
Fetching meta-data from datasource of type "cloud-drive"
Attempting to read from "/media/configvirtfs/openstack/latest/meta_data.json"
Attempting to read from "/media/configvirtfs/openstack"
Failed fetching meta-data from datasource: read /media/configvirtfs/openstack: is a directory
```

I'm using kvm.
Any idea?
Thanks

**Comment by crawford** [comment]
This bug has been fixed in coreos-cloudinit 1.3.1 and will begin rolling out shortly in 592.0.0.

**Comment by aledbf** [comment]
@crawford thanks

**Comment by markuskobler** [comment]
restarting with 592.0.0 looks like the meta_data.json issue is fixed but it still fails to start user-configdrive.service thanks to trying to read meta-data from /media/configdrive/openstack directory.


```
Feb 14 02:03:37 localhost coreos-cloudinit[571]: Fetching meta-data from datasource of type "cloud-drive"
Feb 14 02:03:37 localhost coreos-cloudinit[571]: Attempting to read from "/media/configdrive/openstack/latest/meta_data.json"
Feb 14 02:03:37 localhost coreos-cloudinit[571]: Attempting to read from "/media/configdrive/openstack"
Feb 14 02:03:37 localhost coreos-cloudinit[571]: Failed fetching meta-data from datasource: read /media/configdrive/openstack: is a directory
Feb 14 02:03:37 localhost systemd[1]: user-configdrive.service: main process exited, code=exited, status=1/FAILURE
Feb 14 02:03:37 localhost systemd[1]: Failed to start Load cloud-config from /media/configdrive.
Feb 14 02:03:37 localhost systemd[1]: Unit user-configdrive.service entered failed state.
Feb 14 02:03:37 localhost systemd[1]: user-configdrive.service failed.
```

**Comment by crawford** [comment]
There is a gap in our ability to properly test this. I'm rolling back cloudinit until we can fully test this.

**Comment by crawford** [comment]
This should be fixed in 593.0.0. coreos-cloudinit was rolled back to 1.2.1.
